### ISSUE TITLE
Define the buildRow function for an Entry object in the definition of the Entry object instead of in the HomeList/TripList/SupermarketList widgets

### ISSUE THREAD

**Issue by KnurpsBram** [KnurpsBram/shoplist#18]
It would make it much more readable if we can make a class definition of `ProductEntry` where we define what that entry should look like when shown in a `HomeList` and in a `TripList`in the class definition of `ProductEntry` itself, instead of being distributed over the `HomeList` and `TripList` definitions.

Currently (psuedocode):
```
class MyEntry {
    <some attributes>
    ....
}

class HomeList{
    ListView( <list of buildRow(entry) for all my entries> )
    Widget _buildRow(entry) {
        return ListTile(
            /// when blue button is pressed, change state of HomeList
        )
    }
}

class TripList{
    ListView( <list of buildRow(entry) for all my entries> )
    Widget _buildRow(entry) {
        return ListTile(
            /// when yellow button is pressed, change state of TripList
        )
    }
}
```
what I want
```
class MyEntry {
    <some attributes>
    ....

    Widget buildRowHomeList() {
        return ListTile(
            /// when blue button is pressed, change state of HomeList
        )
    }

    Widget buildRowTripList(entry) {
        return ListTile(
            /// when yellow button is pressed, change state of TripList
        )
    }
}

class HomeList{
    ListView( <list of buildRowHomeList(entry) for all my entries> )
}

class TripList{
    ListView( <list of buildRowTripList(entry) for all my entries> )
}
```

What I'm running into is that I need to define `setState()` for the `onPressed` properties of the buttons in my `ListTile`. To my understanding that `setState()` function refers to the state of the current `State` object, which `HomeListState` and `TripListState` are subclasses of. If I move the ListTile definition out of the definition of my `HomeListState` class I don't know how to make that ListTile refer to the HomeListState it will be placed in.

Cherry on top if I can also move the definition of `Entry` to a different file and import it into the file where `HomeListState` is defined.

**Comment by KnurpsBram** [comment]
According to @akashrajkn advice: It's customary to have the following filestructure:
```
src
  lib
    components/
    models/
    screens/
    util/
    main.dart
```
`components` are visual components, things like a listtile with icons to press.
`models` are things whose attributes don't change when you view them differently, probably `ProductEntry` is a model.
`screens` are things like the homescreen that houses a lot of components

screens should be stateful and components are meant to be stateless. If you want your component to change the state or attribute of a model, you should pass a callback function to it, the callback function should change the state.

These are not laws but more 'rules of thumb'

**Comment by KnurpsBram** [comment]
- **Models** are things that hold data, they are like datatypes or classes. A shoplist entry is a model, it has attributes such as 'isCheckedOff'. The AppData object is also a model
- **Components** are visual components that can be reused, such as a row in a reusablelistview. The component can define that something has a button, but what happens when the button is pressed is defined in a higher-level function (a source code in the screen folder) via a callback function.
- **Screens** encode the list of components that are shown to the user. It reuses the things defined in components and models. What happens when the user interacts is defined in the source code of a Screen; it is then attached to some button or textfield in a lower-level component definition.

Fixed with #27
